### PR TITLE
Add new module digital_ocean_floating_ip_facts

### PR DESCRIPTION
**Pull request by pmarques**
##### SUMMARY

Add module to fetch Digital Ocean Digital Ocean Floating IPs to a droplet.

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

- cloud/digital_ocean/digital_ocean_floating_ip_facts

### CODE REVIEW

**Comment by pmarques** [comment]
he PR with **do_floating_ip** module is https://github.com/ansible/ansible/pull/18922

File: lib/ansible/modules/cloud/digital_ocean/do_floating_ip_facts.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+notes:
+  - Version 2 of DigitalOcean API is used.
+requirements:
+  - "python >= 2.6"
+'''
+
+
+EXAMPLES = '''
+- name: "Create a Floating IP in regigin lon1"
```
Comment (BondAnthony) [inline lib/ansible/modules/cloud/digital_ocean/do_floating_ip_facts.py]: Spelling... region

File: lib/ansible/modules/cloud/digital_ocean/do_floating_ip_facts.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    required: true
+
+notes:
+  - Version 2 of DigitalOcean API is used.
+requirements:
+  - "python >= 2.6"
+'''
+
+
+EXAMPLES = '''
```
Comment (BondAnthony) [inline lib/ansible/modules/cloud/digital_ocean/do_floating_ip_facts.py]: Could you expand on the example and provide a more in-depth how-to?

**Review by BondAnthony — COMMENTED** [review]
needs_revision

File: lib/ansible/modules/cloud/digital_ocean/do_floating_ip_facts.py (hunk 2 above)
Comment (pmarques) [reply lib/ansible/modules/cloud/digital_ocean/do_floating_ip_facts.py]: I will try to check other modules and documentation, but shouldn't this be covered by the return value?

File: lib/ansible/modules/cloud/digital_ocean/do_floating_ip_facts.py (hunk 1 above)
Comment (pmarques) [reply lib/ansible/modules/cloud/digital_ocean/do_floating_ip_facts.py]: fixed

**Comment by BondAnthony** [comment]
@pmarques the openstack modules have a number of facts examples, take a look at os_flavor_facts.

**Comment by BondAnthony** [comment]
shipit

**Review by BondAnthony — APPROVED** [review]

**Comment by BondAnthony** [comment]
shipit

**Comment by BondAnthony** [comment]
@pmarques could you add a couple examples of using the actual fact? Possibly something like the following:

`- debug:
    msg="Floating IP is {{ register.floating_ip.ip }}"
`

**Comment by pmarques** [comment]
@BondAnthony pagination implemented and example updated

**Comment by BondAnthony** [comment]
shipit

**Comment by BondAnthony** [comment]
bot_status

**Comment by BondAnthony** [comment]
@pmarques could you please update your version_added to 2.5.

**Comment by BondAnthony** [comment]
shipit

Thank you @pmarques

**Pull request merged**